### PR TITLE
Dat 3732 - collapsible sections

### PR DESCRIPTION
**Pull request by rybmat**
handling collapsible section on the app side of Infobox Builder

### CODE REVIEW

File: extensions/wikia/PortableInfoboxBuilder/services/Nodes/NodeGroup.class.php [hunk 1]
```diff
@@ -16,7 +16,26 @@ class NodeGroup extends Node {
 	protected $requiredChildNodes = [ 'header' ];
 
 
+	/**
+	 * @var array of string
+	 */
+	protected $allowedAttributes = [ 'collapse' ];
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/services/Nodes/NodeGroup.class.php:22]: We need to have check for supported (allowed) values of this attribute too (also in the tests)

File: extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php [hunk 2]
```diff
@@ -245,12 +246,21 @@ protected function importNodeToDom( $domDocument, $childNodeDom ) {
 	}
 
 	/**
+	 * @param $collapse
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php]: This seems off

File: extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php [hunk 3]
```diff
@@ -245,12 +246,21 @@ protected function importNodeToDom( $domDocument, $childNodeDom ) {
...
 	 * @return DOMElement
 	 */
-	protected function createGroupDom() {
-		return dom_import_simplexml(
-			new SimpleXMLElement( '<' . \Wikia\PortableInfoboxBuilder\Nodes\NodeGroup::XML_TAG_NAME . '/>' )
-		);
+	protected function createGroupDom($childHeader, $childNodeDom) {
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php]: hmm, what do you think about

```
protected function createGroupDom($headerNodeDom, $collapsible) { ... }
```

invoked in 
https://github.com/Wikia/app/pull/9819/files#diff-f997a9b22af5a0ef703dfc816b8678ebR202 
and
https://github.com/Wikia/app/pull/9819/files#diff-f997a9b22af5a0ef703dfc816b8678ebR216
like

```
$currentGroupDom = $this->createGroupDom( $childNodeDom, $this->isCollapsibleHeader($currentChildNode));
```

That would make the createGroupDom more explicit and ensure the share of responsibilities.

File: extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php [hunk 4]
```diff
@@ -245,12 +246,21 @@ protected function importNodeToDom( $domDocument, $childNodeDom ) {
...
+		$groupElem = new SimpleXMLElement( '<' . \Wikia\PortableInfoboxBuilder\Nodes\NodeGroup::XML_TAG_NAME . '/>' );
+		if ( $childHeader['section_collapsible'] ) {
+			$groupElem->addAttribute('collapse', 'open');
+		}
+
+		// clean the mess connected to differences in xml tree and data model on mercury side
+		$childNodeDom->removeAttribute('section_collapsible');
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php]: Please create a new header instead or modify it in the generic flow of getFormattedMarkup in which it is created, as we discussed offline.

File: extensions/wikia/PortableInfoboxBuilder/services/Nodes/NodeGroup.class.php (hunk 1 above)
Comment (rybmat) [inline extensions/wikia/PortableInfoboxBuilder/services/Nodes/NodeGroup.class.php:22]: done

**Comment by sebastianmarzjan** [comment]
Waiting for tests, and LGTM if they pass

**Pull request merged**